omero_marshal/decode/decoders/original_file.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Glencoe Software, Inc. All rights reserved.
+#
+# This software is distributed under the terms described by the LICENCE file
+# you can find at the root of the distribution bundle.
+# If the file is missing please request a copy by contacting
+# jason@glencoesoftware.com.
+#
+
+from .. import Decoder
+from omero.model import OriginalFileI
+
+
+class OriginalFileDecoder(Decoder):
+
+    TYPE = 'TBD#OriginalFile'
+
+    OMERO_CLASS = OriginalFileI
+
+    def decode(self, data):
+        v = super(OriginalFileDecoder, self).decode(data)
+        self.set_property(v, 'path', data.get('path'))
+        self.set_property(v, 'size', data.get('size'))
+        self.set_property(v, 'atime', data.get('atime'))
+        self.set_property(v, 'mtime', data.get('mtime'))
+        self.set_property(v, 'ctime', data.get('ctime'))
+        self.set_property(v, 'hash', data.get('hash'))
+        self.set_property(v, 'mimetype', data.get('mimetype'))
+        self.set_property(v, 'name', data.get('name'))
```

Comment:
Since `OriginalFile` has no analogue in the OME data model and thus essentially an internal object to OMERO I have kept the same naming style as the other internal OMERO objects which *should* reflect the field naming convention of OMERO itself. These include `Details`, `Permissions`, all the enumerations, etc.

Happy to be convinced otherwise.